### ISSUE TITLE
Can you change the backbone model to something else than T5? LLama?

### ISSUE THREAD

**Issue by hohoCode** [tum-ai/number-token-loss#52]
Great work!

Quick question: Can we use Llama instead for experiments? Or have you done such experiments replacing T5 with Llama or others? How is the performance if using Llama (or others) given all other conditions same?

Thanks!

**Comment by jannisborn** [comment]
Hi @hohoCode,

thanks for the interest in our work. Yes, we are actively working on this and will soon release a MWE to train pretty much any LM with NTL. The code in this repo is the dev-code. It's fully functional, so you should be able to adapt it, but it's a bit geared to support evaluation etc. Since this paper is WIP, we havent arrived at the most lightweight implementation that we would like to ship to the community. We have done some initial experiments with the [Granite family](https://huggingface.co/ibm-granite) and saw mild benefits of NTL. I'll leave this open and we'll follow up here once we have an update!
Hope this helps

**Comment by Larspennig** [comment]
Hi @hohoCode,

we just added the MWE to the codebase – you can find it under:
`number-token-loss/src/ntl/loss_functions/base_number_token_loss.py`

While training a Llama model using this loss function should work, please note that we have not yet completed a full fine-tuning run with this model due to limited resources at this time.

We hope this eases the loss integration for you!

**Comment by jannisborn** [comment]
Great! Closing this as completed but feel free to reopen if applicable @hohoCode
